tests/compile-test.rs

```diff
@@ -355,10 +355,42 @@ fn run_ui_cargo() {
 #[test]
 fn compile_test() {
     set_var("CLIPPY_DISABLE_DOCS_LINKS", "true");
-    run_ui();
-    run_ui_toml();
-    run_ui_cargo();
-    run_internal_tests();
+    // The SPEEDTEST_* env variables can be used to check Clippy's performance on your PR. It runs the
+    // affected test 1000 times and gets the average.
+    if let Ok(speedtest) = std::env::var("SPEEDTEST") {
+        println!("----------- STARTING SPEEDTEST -----------");
+        let f;
+        match speedtest.as_str() {
+            "ui" => f = run_ui as fn(),
+            "cargo" => f = run_ui_cargo as fn(),
+            "toml" => f = run_ui_toml as fn(),
+            "internal" => f = run_internal_tests as fn(),
+
+            _ => panic!("unknown speedtest: {speedtest} || accepted speedtests are: [ui, cargo, toml, internal]"),
+        }
+
+        let iterations;
+        if let Ok(iterations_str) = std::env::var("SPEEDTEST_ITERATIONS") {
+            iterations = iterations_str
+                .parse::<u64>()
+                .unwrap_or_else(|_| panic!("Couldn't parse `{}`, please use a valid u64", iterations_str));
+        } else {
+            iterations = 1000;
+        }
+
+        let mut sum = 0;
+        for _ in 0..iterations {
+            let start = std::time::Instant::now();
+            f();
+            sum += start.elapsed().as_millis();
+        }
```

Comment:
since the timing itself also takes some time to run, would it make sense to move the timing stuff out of the loop and divide the time taken by the number of iterations after the loop? the Instant::now().elapsed() overhead might also be negligible compared to how long a lint pass takes, so maybe this doesn't matter that much (from what I remember, it was around 40ns on the playground...)

Reply:
I think the usual way to micro benchmark is to time variable numbers of iterations, then do a linear regression, but this is clearly a macro benchmark, so don't worry about either loop or time measurement overhead.